source/PlayState.hx

```diff
@@ -1264,7 +1272,72 @@ class PlayState extends MusicBeatState
 
 		super.update(elapsed);
 
-		scoreTxt.text = "Score:" + songScore;
+		scoreTxt.text = "Score: " + songScore;
+
+		//im so sorry for this - i know that there might a better way to do this, but it works lol
+		switch(curStage) {
+			case 'stage':
+				switch(SONG.song.toLowerCase()) {
+					case 'tutorial':
+						curSongText.text = "Song: " + SONG.song.toLowerCase();
+					case 'bopeebo':
+						curSongText.text = "Song: " + SONG.song.toLowerCase();
+					case 'fresh':
+						curSongText.text = "Song: " + SONG.song.toLowerCase();
+					case 'dadbattle':
+						curSongText.text = "Song: " + SONG.song.toLowerCase();
+
+				}
+			case 'spooky':
+				switch(SONG.song.toLowerCase()) {
+					case 'spookeez':
+						curSongText.text = "Song: " + SONG.song.toLowerCase();
+					case 'south':
+						curSongText.text = "Song: " + SONG.song.toLowerCase();
+				}
+			case 'philly':
+				switch(SONG.song.toLowerCase()) {
+					case 'pico': 
+						curSongText.text = "Song: " + SONG.song.toLowerCase();
+					case 'blammed':
+						curSongText.text = "Song: " + SONG.song.toLowerCase();
+					case 'philly': 
+						curSongText.text = "Song: " + SONG.song.toLowerCase();
+				}
+			case 'limo':
+				switch(SONG.song.toLowerCase()) {
+					case 'satin-panties':
+						curSongText.text = "Song: " + SONG.song.toLowerCase();
+					case 'high':
+						curSongText.text = "Song: " + SONG.song.toLowerCase();
+					case 'milf':
+						curSongText.text = "Song: " + SONG.song.toLowerCase();
+				}
+			case 'mall':
+				switch(SONG.song.toLowerCase()){
+					case 'cocoa':
+						curSongText.text = "Song: " + SONG.song.toLowerCase();
+					case 'eggnog':
+						curSongText.text = "Song: " + SONG.song.toLowerCase();
+				}
+			case 'mallEvil':
+				switch(SONG.song.toLowerCase()) {
+					case 'winter-horrorland':
+						curSongText.text = "Song: " + SONG.song.toLowerCase();
+				}
+			case 'school':
+				switch(SONG.song.toLowerCase()) {
+					case 'senpai':
+						curSongText.text = "Song: " + SONG.song.toLowerCase();
+					case 'roses':
+						curSongText.text = "Song: " + SONG.song.toLowerCase();
+				}
+			case 'schoolEvil':
+				switch(SONG.song.toLowerCase()){
+					case 'thorns':
+						curSongText.text = "Song: " + SONG.song.toLowerCase();
+				}
+		}
```

Comment:
```suggestion
		curSongText.text = "Song: " + SONG.song.toLowerCase();
```
The value of the `text` property of `curSongText` is always changed to the exact same thing on that huge `switch` statement, so why not just do this?

Reply:
I tried it like that, didn't work for all the dad songs for some reason.

Reply:
update: holy shit you fucking destroyed me LOL

idk why it didn't work for me before, I was probably storing data to the text on the wrong function or something,
excuse my stupidity lol still new to haxeflixel.